### DISCUSSION TITLE
Export mask to PNG image file

### DISCUSSION THREAD

**Discussion by leojroche** [gdsfactory/gdsfactory#1629] [General]
Hello everyone, 

How feasible would it be to export a mask generated by Gdsfactory to an image file (with a specific resolution), for instance a PNG file? 

Thanks!

**Comment by jan-david-fischbach** [comment]
Have you considered using the matplotlib plotting functionality for this purpose?

```python
c.plot_matplotlib()
```

**Comment by joamatab** [comment]
I would use `plot_klayout` instead

what do you think of deprecating the matplotlib plotter? as it can handle big layouts with many polygons

see https://github.com/gdsfactory/gdsfactory/pull/1953

**Comment by jan-david-fischbach** [comment]
Jep for big designs that is the better option. I was thinking more in terms of single device images for presentations and the like. For this application I typically find matplotlib to have a 'cleaner' output. 

Maybe outputting to some vector formats (svg/pdf) is also nice?

**Reply by tvt173** [reply]
gdstk can also write to SVG
https://heitzmann.github.io/gdstk/gettingstarted.html#saving-a-layout-file

**Comment by joamatab** [comment]
We added the docs here

```python
fig = c.plot_klayout()
fig.savefig("demo.png")
```

https://gdsfactory.github.io/gdsfactory/notebooks/00_geometry.html#write
